### PR TITLE
DO NOT MERGE - Drillview wix animations

### PR DESCRIPTION
**Pull request by ronenst**
### What changed

Replaced internal `SlideAnimation` with `wix-animations` 

### Why it changed

If a sidebar transition is invoked while the sidebar is already in transition the sidebar can reach a state where the menu disappears completely.

In addition a story + fixes for RTL were added, the `rtl` prop is now required for the `DrillView` to slide in the correct direction.

---

- [ ] Change is tested
- [x] Change is documented
- [x] Build is green
- [ ] Change of UX is approved by [wuwa](https://github.com/wuwa) or [milkyfruit](https://github.com/milkyfruit)

### Thanks for contributing :)

My pleasure ;)

### CODE REVIEW

File: src/SideMenu/core/styles.scss [hunk 1]
```diff
@@ -13,6 +13,10 @@
   flex-grow: 1;
 }
 
+.root:global(.rtl) {
```
Comment (argshook) [inline src/SideMenu/core/styles.scss]: select attribute `.root:global([dir="rtl"])`

File: stories/SideMenu/ExampleSideMenuDrillRtl.js [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        <SideMenu.FooterLink
+          href="https://support.wix.com/"
+          target="_blank"
+          icon={<HelpIcon size="1em"/>}
+          >
+          Help Me!
+        </SideMenu.FooterLink>
+
+        <SideMenu.FooterTinyLink
+          href="https://support.wix.com/en/article/wix-seo-wiz-suggestions-and-feedback"
```
Comment (argshook) [inline stories/SideMenu/ExampleSideMenuDrillRtl.js]: maybe `https://support.wix.com/` here also

File: stories/SideMenu/ExampleSideMenuDrillRtl.js [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      items
+    };
+  }
+
+  componentWillMount() {
+    items[2].items[1].isActive = true;
```
Comment (argshook) [inline stories/SideMenu/ExampleSideMenuDrillRtl.js]: i this section really needed for storybook example?

File: src/SideMenu/core/navigation/styles.scss [hunk 4]
```diff
@@ -160,6 +160,11 @@ $padding-vertical: 14px;
   padding: 17px 26px;
   top: 0;
   right:0;
+}
 
-
+:global(.rtl) {
```
Comment (argshook) [inline src/SideMenu/core/navigation/styles.scss]: select attribute `:global([dir="rtl"])`

**Comment by ronenst** [comment]
@argshook Thanks for the review, I've attempted to address the issues.

I only now noticed there is an issue where the animation is reversed the first time a component is mounted, I guess I didn't see it before because I was relying on hot reloading.

Please don't merge yet until I can fix it.

**Comment by ronenst** [comment]
Closing this and we'll try another solution, this is very hard to achieve `wix-animations` without major changes. In `wix-animations` the "exit" direction of the animation must be specified when the component is mounted but for submenus which have submenus we don't know if the direction until the user decides to click either the back button or the parent item, therefore the direction is sometimes wrong.

**Pull request closed without merging**